manim/__main__.py

```diff
@@ -1,111 +1,33 @@
+    default="render",
+    no_args_is_help=True,
+    context_settings=CONTEXT_SETTINGS,
+    help="Animation engine for explanatory math videos",
+    epilog=EPILOG,
 )
-from manim.utils.file_ops import open_file as open_media_file
-from manim._config.main_utils import parse_args
-
-
-def open_file_if_needed(file_writer):
-    if config["verbosity"] != "DEBUG":
-        curr_stdout = sys.stdout
-        sys.stdout = open(os.devnull, "w")
-
-    open_file = any([config["preview"], config["show_in_file_browser"]])
-
-    if open_file:
-        file_paths = []
-
-        if config["save_last_frame"]:
-            file_paths.append(file_writer.image_file_path)
-        if config["write_to_movie"] and not config["save_as_gif"]:
-            file_paths.append(file_writer.movie_file_path)
-        if config["save_as_gif"]:
-            file_paths.append(file_writer.gif_file_path)
-
-        for file_path in file_paths:
-            if config["show_in_file_browser"]:
-                open_media_file(file_path, True)
-            if config["preview"]:
-                open_media_file(file_path, False)
-
-    if config["verbosity"] != "DEBUG":
-        sys.stdout.close()
-        sys.stdout = curr_stdout
-
-
-def main():
-    console.print(f"Manim Community [green]v{__version__}[/green]")
-    args = parse_args(sys.argv)
-
-    if hasattr(args, "cmd"):
-        if args.cmd == "cfg":
-            if args.subcmd:
-                from manim._config import cfg_subcmds
-
-                if args.subcmd == "write":
-                    cfg_subcmds.write(args.level, args.open)
-                elif args.subcmd == "show":
-                    cfg_subcmds.show()
-                elif args.subcmd == "export":
-                    cfg_subcmds.export(args.dir)
-            else:
-                logger.error("No subcommand provided; Exiting...")
-
-        elif args.cmd == "plugins":
-            from manim.plugins import plugins_flags
-
-            if args.list:
-                plugins_flags.list_plugins()
-            elif not args.list:
-                logger.error("No flag provided; Exiting...")
-
-        # elif args.cmd == "some_other_cmd":
-        #     something_else_here()
-
-    else:
-        config.digest_args(args)
-        input_file = config.get_dir("input_file")
-
-        if config["use_opengl_renderer"]:
-            from manim.renderer.opengl_renderer import OpenGLRenderer
-
-            for SceneClass in scene_classes_from_file(input_file):
-                try:
-                    renderer = OpenGLRenderer()
-                    scene = SceneClass(renderer)
-                    scene.render()
-                except Exception:
-                    console.print_exception()
-        elif config["use_webgl_renderer"]:
-            try:
-                from manim.grpc.impl import frame_server_impl
+@click.version_option(
+    version=__version__, prog_name="Manim", message="%(prog)s v%(version)s"
+)
+@click.pass_context
+def manim(ctx):
```

Comment:
The reason was because of `test_commands.py`. Whatever the entry point function is named gets shown in the [resulting output](https://github.com/jsonvillanueva/manim/blob/cde92bcb8207d6e3b450d4a32d124c098a2e8218/tests/test_commands.py#L22). In order to make the resulting output match the expected output from the CLI, I made the change here. Otherwise, the expected output in `test_commands.py` would have to be changed to:

>    expected_output = """Usage: main cfg [OPTIONS] COMMAND [ARGS]...

Reply:
I don't think it will do so, there must be an option to change these if it does so, and I don't like to have a function name same as the module name.

Reply:
Well I'm currently uncertain how to adjust this, but I'll look into it. In the meantime, I've applied all the other suggestions.

Reply:
I reverted the change and updated the test_commands test cases to just have main. It's unimportant for this PR that it matches, but I'll still look into whether it can be changed with Click's CliRunner